pkg/util/graph/backup_graph.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of the Kubevirt Velero Plugin project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Velero Plugin Authors.
+ *
+ */
+
+package vmgraph
+
+import (
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	v1 "kubevirt.io/api/core/v1"
+)
+
+// NewVirtualMachineBackupGraph returns the backup object graph for a specific VM
+func NewVirtualMachineBackupGraph(vm *v1.VirtualMachine) ([]velero.ResourceIdentifier, error) {
+	var resources []velero.ResourceIdentifier
+	var err error
+	namespace := vm.GetNamespace()
+
+	if vm.Spec.Instancetype != nil {
+		resources = addInstanceType(*vm.Spec.Instancetype, vm.GetNamespace(), resources)
+	}
+	if vm.Spec.Preference != nil {
+		resources = addPreferenceType(*vm.Spec.Preference, vm.GetNamespace(), resources)
+	}
+	if vm.Status.Created {
+		resources = addVeleroResource(vm.GetName(), namespace, "virtualmachineinstances", resources)
+		// Returning full backup even if there was an error retrieving the launcher pod.
+		// The caller can decide wether to use the backup without launcher pod or handle the error.
+		resources, err = addLauncherPod(vm.GetName(), vm.GetNamespace(), resources)
+	}
+
+	return addCommonVMIObjectGraph(vm.Spec.Template.Spec, namespace, true, resources), err
+}
+
+// NewVirtualMachineInstanceBackupGraph returns the backup object graph for a specific VMI
+func NewVirtualMachineInstanceBackupGraph(vmi *v1.VirtualMachineInstance) ([]velero.ResourceIdentifier, error) {
+	var resources []velero.ResourceIdentifier
+	var err error
+	// Returning full backup even if there was an error retrieving the launcher pod.
+	// The caller can decide wether to use the backup without launcher pod or handle the error.
+	resources, err = addLauncherPod(vmi.GetName(), vmi.GetNamespace(), resources)
+	return addCommonVMIObjectGraph(vmi.Spec, vmi.GetNamespace(), true, resources), err
+}
```

Comment:
Wonder if DataVolume should be an entrypoint as well?  Basically anything that has a a `*_backup_item_action.go`?

Reply:
Would that only include the pvc? maybe the populator source if it has one too? I can add it but seems a bit unnecessary, but I don't mind having it

Reply:
> Would that only include the pvc?

I think so?  Basically whatever the "extra" items we add to the backup I don't think we include populator sources.  That seems like base functionality that could/should be added to velero core

Basically I think there should be a graph function for each of our types that we specially handle

Reply:
Added commit with datavolume backup graph